### PR TITLE
Add manual network self-check

### PR DESCRIPTION
**Pull request by SHIINASAMA**
## Summary

- add a shared OSS/Pro manual Network Self-Check page
- run ordered connectivity, DNS resolution, and system proxy checks
- present Normal, Abnormal, and Indeterminate results in an adaptive full-width macOS workbench
- keep diagnostics user initiated with a minimum visible duration per check
- use outcome-oriented localized result copy without exposing test domains or proxy endpoints
- reduce duplicate debug console output and remove noisy WindowAccessor logging

## User impact

Users can quickly determine whether their Mac has a usable network connection, working DNS resolution, or a stale system proxy configuration. Results stay inline in the current page and adapt to window size without opening another dialog.

## Implementation

- `NWPathMonitor` for connectivity state
- `DNSServiceGetAddrInfo` for DNS resolution
- `CFNetworkCopySystemProxySettings` plus bounded TCP reachability checks for explicit proxies
- unified `DiagnosticCheck` and ordered `DiagnosticRunner`
- shared source composition for OSS and Pro
- English, German, Spanish, Japanese, and Simplified Chinese localization

## App Store follow-up

- track a pre-submission TODO to expand `NSLocalNetworkUsageDescription` so it also describes checking a configured local proxy

## Validation

- focused network diagnostics tests
- complete `WiFiLensTests` unit test bundle
- `WiFi Lens` Debug build
- `WiFi Lens Pro` Debug build
- localization catalog JSON and completeness scan
- `git diff --check`

### CODE REVIEW

File: WiFiLens/Sources/WiFiLens/App/Logging.swift [hunk 1]
```diff
@@ -66,6 +66,17 @@ enum AppLogger {
 
 // MARK: - ConsoleLogHandler
 
+enum DebugConsoleLogPolicy {
+    static func shouldWrite(_ level: Logging.Logger.Level) -> Bool {
+        switch level {
+        case .trace, .debug:
+            true
+        case .info, .notice, .warning, .error, .critical:
```
Comment (SHIINASAMA) [inline WiFiLens/Sources/WiFiLens/App/Logging.swift]: Console now suppresses `info`/`notice`/`warning`/`error`/`critical` on stdout, keeping only `trace`/`debug`. This drops `error`/`critical` from stdout even in Debug builds, so local xcodebuild/Xcode console debugging won't surface errors. De-dup vs OSLog usually targets `info`/`notice` only — was suppressing `warning`+ intended? Consider `case .warning, .error, .critical: true`.

File: WiFiLens/Sources/WiFiLens/NetworkDiagnostics/NetworkDiagnosticsViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        activeTask = Task { [weak self] in
+            let results = await runner.run { [weak self] result in
+                await self?.accept(result)
+            }
+            self?.finish(results)
+        }
+        return true
+    }
+
+    func cancel() {
```
Comment (SHIINASAMA) [inline WiFiLens/Sources/WiFiLens/NetworkDiagnostics/NetworkDiagnosticsViewModel.swift]: `cancel()` has no call site in production or tests, and the view has no cancel entry point (it only disables the button while running). Either expose a cancel button in `NetworkDiagnosticsView` or remove this method (YAGNI). `deinit`'s `activeTask?.cancel()` remains effective regardless.

File: WiFiLens/Sources/WiFiLens/NetworkDiagnostics/DNSResolutionCheck.swift [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            }
+
+            context.install(serviceRef: serviceRef)
+            DNSServiceSetDispatchQueue(
+                serviceRef,
+                DispatchQueue(label: "io.github.kaoru.wifi-lens.network-diagnostics.dns")
+            )
+
+            Task {
+                try? await Task.sleep(for: timeout)
```
Comment (SHIINASAMA) [inline WiFiLens/Sources/WiFiLens/NetworkDiagnostics/DNSResolutionCheck.swift:52]: On early resolution this detached timeout task still sleeps the full timeout (5s) before calling `finish` (a no-op under the lock). Not a bug, but the task lingers holding the context. Consider cancelling it once the continuation resumes (store the handle and cancel in `finish`, or model timeout via a task group). Same pattern in SystemProxyCheck.swift:125.

**Review by SHIINASAMA — COMMENTED** [review]
## Code review — manual network self-check

Overall this is in good shape: clean architecture, thorough dependency injection, correct concurrency handling (lock-guarded continuations prevent double-resume), solid test coverage, and complete localization (33 new keys across all 5 languages). No blocking bugs found. A few points to consider below (1 worth confirming, 3 minor / non-blocking).

### Worth confirming
- **Console log policy now suppresses `error`/`critical`** (`Logging.swift`). `DebugConsoleLogPolicy.shouldWrite` only emits `trace`/`debug` to stdout. The stated goal is de-duplication (these levels also go to OSLog), but the side effect is that `error`/`critical` no longer print to stdout even in Debug builds — so local `xcodebuild` / Xcode console debugging won't surface error lines. De-duplication usually targets `info`/`notice`; suppressing `warning`+ may be more than intended. Intentional?

### Minor (non-blocking)
- **`NetworkDiagnosticsViewModel.cancel()` is dead code (YAGNI).** No production or test call site. During `running` the view only `.disabled`s the button — there's no cancel entry point. `deinit`'s `activeTask?.cancel()` is effective, but the public `cancel()` is unused. Either expose a cancel button in the UI or drop the method. (`waitForCompletion()` being test-only is fine — it's a legitimate test hook.)
- **Timeout `Task { sleep }` isn't cancelled on early completion** (`DNSResolutionCheck`, `SystemProxyCheck`). When the operation finishes early, the detached timeout task still sleeps the full timeout (DNS 5s / proxy 3s) before calling `finish` (a no-op thanks to the lock). Minor resource retention, not a bug — could be cleaned up with a task group or by cancelling a stored handle in `finish`.
- **Hardcoded `example.com` for the DNS probe.** Reasonable choice (IANA-reserved, resolvable) and there's a test ensuring it never leaks into UI copy. Just noting the edge case: offline-with-cached-DNS could theoretically mislead. Acceptable tradeoff.

### FYI
- `CLAUDE.md` still states localization is `en/ja/zh-Hans`, but the catalog is actually 5 languages (`de`/`es` predate this PR). Since this PR touches the xcstrings, that doc line could be updated in a follow-up.

File: WiFiLens/Sources/WiFiLens/App/Logging.swift (hunk 1 above)
Comment (SHIINASAMA) [reply WiFiLens/Sources/WiFiLens/App/Logging.swift]: Fixed in e379c1f. Debug stdout now suppresses only info and notice; warning, error, and critical remain visible. The regression test now covers every affected level.

File: WiFiLens/Sources/WiFiLens/NetworkDiagnostics/NetworkDiagnosticsViewModel.swift (hunk 2 above)
Comment (SHIINASAMA) [reply WiFiLens/Sources/WiFiLens/NetworkDiagnostics/NetworkDiagnosticsViewModel.swift]: Removed the unused public cancel() method in e379c1f. Task cancellation on view-model deinitialization remains unchanged.

File: WiFiLens/Sources/WiFiLens/NetworkDiagnostics/DNSResolutionCheck.swift (hunk 3 above)
Comment (SHIINASAMA) [reply WiFiLens/Sources/WiFiLens/NetworkDiagnostics/DNSResolutionCheck.swift:52]: Deferring this non-blocking cleanup. The checks are manual and bounded to 3–5 seconds, while the lock prevents double resume. A cancellable timeout abstraction would add complexity without changing current user-visible behavior.
